contracts/examples/VCPayment-v2.sol

```diff
@@ -0,0 +1,276 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.20;
+import {Ownable2StepUpgradeable} from "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+
+contract VCPaymentV2 is Ownable2StepUpgradeable {
+    /**
+     * @dev Version of contract
+     */
+    string public constant VERSION = '2.0.0';
+
+    /// @custom:storage-location erc7201:iden3.storage.VCPayment
+    struct PaymentData {
+        uint256 issuerId;
+        uint256 schemaHash;
+        uint256 valueToPay;
+        uint256 ownerPartPercent;
+        address withdrawAddress;
+        // for reporting
+        uint256 totalValue;
+    }
+
+    /**
+     * @dev Main storage structure for the contract
+     */
+    struct VCPaymentStorage {
+        /**
+         * @dev mapping of paymentDataId - keccak256(abi.encode(issuerId, schemaHash)) => PaymentData
+         */
+        mapping(bytes32 paymentDataId => PaymentData paymentData) paymentData;
+
+        /**
+         * @dev mapping of paymentRequestId - keccak256(abi.encode(issuerId, paymentId)) => bool
+         */
+        mapping(bytes32 paymentRequestId => bool isPaid) payments;
+
+        /**
+         * @dev mapping of issuerAddress - balance
+         */
+        mapping(address issuerAddress => uint256 balance) issuerAddressBalance;
+
+        /**
+         * @dev owner balance
+         */
+        uint256 ownerBalance;
+
+        /**
+         * @dev list of paymentDataId - keccak256(abi.encode(issuerId, schemaHash))
+         */
+        bytes32[] paymentDataIds;
+    }
+
+    // keccak256(abi.encode(uint256(keccak256("iden3.storage.VCPayment")) - 1)) &
+    //    ~bytes32(uint256(0xff));
+    bytes32 private constant PaymentDataStorageLocation =
+        0xbb49acb92ce91902600caabfefad66ed7ac2a150edbd631ab48a5501402b3300;
+
+    function _getPaymentDataStorage() private pure returns (VCPaymentStorage storage $) {
+        assembly {
+            $.slot := PaymentDataStorageLocation
+        }
+    }
```

Comment:
you can rename `PaymentDataStorageLocation` to `VCPaymentStorageLocation`
and `_getPaymentDataStorage` to `_getVCPaymentStorage`